x/auction/keeper/auction.go

```diff
@@ -180,6 +181,31 @@ func (k Keeper) ValidateAuctionBundle(bidder sdk.AccAddress, bundleSigners []map
 	return nil
 }
 
+// ValidateBundleTimeouts validates that the timeouts of the transactions in the bundle are valid. We consider
+// the timeouts valid iff all of the bidders txs in the bundle are equal to the timeout of the bid. This is
+// mandatory as otherwise other searchers can potentially include other searcher's transactions in their bundles.
+func (k Keeper) ValidateBundleTimeouts(bidInfo *types.BidInfo) error {
+	bidder := bidInfo.Bidder.String()
+	bidTimeout := bidInfo.Timeout
+
+	for index, bundleTimeout := range bidInfo.TransactionTimeouts {
+		signers := bidInfo.Signers[index]
```

Comment:
nit: comment here would be nice, spent some time tryna figure out what this did (check if the bidder is signer of a sub-tx). Maybe I'm stupid tho so its fine